File: CRPL.Web/ClientApp/src/app/_Models/Applications/CopyrightRegistrationViewModel.ts
import {ApplicationViewModel} from "./ApplicationViewModel";
import {OwnershipStake} from "../StructuredOwnership/OwnershipStake";
import { CopyrightType } from "../CopyrightType";
import {WorkType} from "../WorkType";
import {Protections} from "./Protections";

export interface CopyrightRegistrationViewModel extends ApplicationViewModel
{
  WorkType: WorkType;
  Title: string;
  WorkHash: string;
  WorkUri: string;
  Legal: string;
  CopyrightType: CopyrightType;
  OwnershipStakes: OwnershipStake[];
  YearsExpire?: number;
  Protections: Protections;
}

File: CRPL.Data/Applications/ViewModels/CopyrightRegistration/CopyrightRegistrationViewModel.cs
using CRPL.Contracts.Structs;
using CRPL.Data.StructuredOwnership;

namespace CRPL.Data.Applications.ViewModels;

public class CopyrightRegistrationViewModel : ApplicationViewModel
{
    public string Title { get; set; }
    public byte[] WorkHash { get; set; }
    public string WorkUri { get; set; }
    public string Legal { get; set; }
    public WorkType WorkType { get; set; }
    public int YearsExpire { get; set; }
    public List<OwnershipStake> OwnershipStakes { get; set; }
    public Protections Protections { get; set; }
}

public class CopyrightRegistrationViewModelWithoutAssociated : ApplicationViewModelWithoutAssociated
{
    
}